src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -184,6 +180,12 @@ internal void Start()
                     throw Interop.GetExceptionForIoErrno(Interop.Sys.GetLastErrorInfo(), _fullDirectory, true);
                 }
 
+                // Normalize the _fullDirectory path to have a trailing slash
+                if (_fullDirectory[_fullDirectory.Length - 1] != '/')
```

Comment:
Yes, the call to ```RealPath``` should ensure that; the call to ```realpath``` [returns null](http://linux.die.net/man/3/realpath) on error, so if the path is not a valid path and cannot be realized to an actual path, the function will return null and we will throw in the if statement above